flowmachine/flowmachine/features/spatial/distance_matrix.py

```diff
@@ -71,6 +73,25 @@ def __init__(self, level="versioned-cell", date=None, return_geometry=False):
 
         super().__init__()
 
+    @property
+    def column_names(self) -> List[str]:
+        cols = get_columns_for_level(self.level)
+
+        try:
+            cols.remove("lat")
+            cols.remove("lon")
+        except ValueError:
+            pass  # Nothing to remove
+
+        col_names = [f"{c}_from" for c in cols]
+        col_names += [f"{c}_to" for c in cols]
+        col_names += [f"{c}_from" for c in ("lon", "lat")]
```

Comment:
Just to check: the lon/lat columns may be missing from the columns (depending on the value of `self.level`), but here we are adding them unconditionally. Is this intentional?

Reply:
lon & lat will always be in the cols returned by `get_columns_for_level` because the only permitted levels include them. This slightly odd dance is to make sure the list returned is in the right order.

This is another class that could use a refactor really - aside from anything else, much of the logic could be replaced by passing in a `VersionedInfrastructure` query.

Reply:
So why do we need the try/except statement if `cols` (obtained from `get_columns_for_level`) will always contain `lat` and `lon`? Under what conditions can the ValueError actually occur?

Reply:
Hah. Good point.